include/private/tdyconstants.h

```diff
@@ -0,0 +1,18 @@
+#if !defined(TDYCONSTANTS_H)
+#define TDYCONSTANTS_H        
+
+#define GRAVITY_CONSTANT              9.8068
```

Comment:
The other mechanism that C offers for defining constants like this is the `static const` declaration, which adds type safety but precludes optimizations in some cases (not these):

```
static const PetscReal GRAVITY_CONSTANT = 9.8068;
```

Various people have various opinions about this kind of stuff. I don't think it matters much as long as we are consistent. But the `static const` method does type checking, which can be nice.

Reply:
I have no objection. `#define GRAVITY_CONSTANT ((PetscReal)9.8068)` is yet another way. In situations like this, there's usually no computational advantage to having it be a compile-time constant, and there are times for debugging that turning off gravity or doubling gravity can be convenient. If constants are controllable via run-time options, then the default constants never need to appear in a header, just in the compilation unit that initializes all the default values.

Reply:
@jeff-cohere @jedbrown, I agree that adding the ability to change gravity and other default values from the command line is a good feature. Is this a reasonable approach https://github.com/TDycores-Project/TDycore/pull/168/commits/6a0038922efd26b693c177a5cef712ca5d60a5e3 to avoid defining values as compile-time constants? 

After this discussion, I'm inclined to consolidate the reading of all the command line options in a single function and save them in a new `tdy->options` struct. How does this sound?

Reply:
That sounds good to me, provided that the options are structured in a way that can be understood when we consolidate our demos to a single driver.